var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -76,7 +79,7 @@ class Mpich(AutotoolsPackage):
 
     depends_on('libfabric', when='netmod=ofi')
 
-    depends_on('libpciaccess')
+    depends_on('libpciaccess', when="+pci")
```

Comment:
@adamqc If you think MacOS users would never need `libpciaccess` for `mpich`, you can condition the when clause on the platform too:

`depends_on('libpciaccess', when="platform=darwin")`

In which case the variant is not needed

Reply:
Sorry for the delayed reply. I think it should be
 ```depends_on('libpciaccess', when="platform!=darwin")```
But I get `==> Error: Invalid character`, it seems that we can not use `!=` in `when` argument. Do you have any suggestions?

Reply:
> I think it should be
> `depends_on('libpciaccess', when="platform!=darwin")`
> But I get `==> Error: Invalid character`, it seems that we can not use `!=` in `when` argument.

You are correct: currently "not" is not supported in Spack spec syntax constraints, and I got the constraint wrong.

So for now I think you have the best solution